### PR TITLE
Design 1.0

### PR DESCRIPTION
**Pull request by kathleenconway**
Just a good old-fashioned setup assistant
- Outbound connection
- Contact mapping
- Debug log management
- Data sync scheduler
- Contact sync configuration

### CODE REVIEW

File: force-app/main/default/lwc/outboundConnectionStep/outboundConnectionStep.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { LightningElement } from 'lwc'
+
+export default class OutboundConnectionStep extends LightningElement {
+    loading = false
```
Comment (sean-appiphony) [inline force-app/main/default/lwc/outboundConnectionStep/outboundConnectionStep.js:4]: missing semicolon

File: force-app/main/default/lwc/batchSchedulerStep/batchSchedulerStep.html [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                        </c-picklist>
+                    </template>
+                    <template if:false={onceDaily}>
+                        <c-picklist
+                            disabled
+                            label="Job Start Time"
+                            options={nonDailyJobStartHourOptions}
+                            value="asap">
+                        </c-picklist>
+                    </template> -->
```
Comment (sean-appiphony) [inline force-app/main/default/lwc/batchSchedulerStep/batchSchedulerStep.html:44]: Will these be brought back at some point? If not, consider removing this comment block

File: force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.js [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+            },
+        ]
+    }
+
+    @track objectValue = 'account'
+    @track objectLabel = 'Account'
+
+    updateObjectValue(event){
+        this.value = event.detail.value;
+        this.value = objectValue;
```
Comment (sean-appiphony) [inline force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.js:49]: Won't this just set `this.value` attribute twice each time?

File: force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.js [hunk 4]
```diff
@@ -0,0 +1,286 @@
+import { LightningElement, api, wire, track } from 'lwc';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+import saveData from '@salesforce/apex/setupAssistant.saveData';
+import getData from '@salesforce/apex/setupAssistant.getData';
+
+export default class FieldMappingStep extends LightningElement {
```
Comment (sean-appiphony) [inline force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.js:6]: Generally speaking, consider grouping up similar elements within the document: attributes with attributes, getters/ setters with getters/ setters, methods with methods. Sometimes it makes sense to group certain blocks of demo code together, but this file appears a little random. Consider cleaning up for code clarity

**Review by sean-appiphony — COMMENTED** [review]
A few things to look for throughout: 
1) Make sure all net new JS files have semicolons at the ends of statements
2) double check your class scoping prefixes to make sure they are consistent throughout the project

File: force-app/main/default/lwc/batchSchedulerStep/batchSchedulerStep.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/batchSchedulerStep/batchSchedulerStep.html:44]: Axed it

File: force-app/main/default/lwc/batchSchedulerStep/batchSchedulerStep.css [hunk 5]
```diff
@@ -0,0 +1,4 @@
+.bss-tooltip {
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/batchSchedulerStep/batchSchedulerStep.css:1]: All but one of these needed to be cleaned out anyway so I changed it to a rl- prefix

File: force-app/main/default/lwc/customPropertyEditor/customPropertyEditor.html [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<template>
+    <lightning-input
+        required
+        label="Record ID"
+        value={recordID}>
+    </lightning-input>
+    <!-- <lightning-input
+        label="Object API Name"
+        value={objectAPI}>
+    </lightning-input> -->
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/customPropertyEditor/customPropertyEditor.html:10]: 🪓

File: force-app/main/default/lwc/debugLogStep/debugLogStep.html [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        onnext={handleNext}
+    >
+        <template if:true={isLoading}>
+            <lightning-spinner alternative-text="spinner"></lightning-spinner>
+        </template>
+            <div slot="description">
+                <div class="slds-var-p-horizontal_large">
+                    <p class="slds-var-m-bottom_x-small">Your Rightsline Salesforce integration automatically creates activity logs for use in the debugging process.</p>
+                    <p>You can choose to log both success and error activity, or error activity alone.</p>
+                    <!-- <p>To improve app performance, you can set a limit on the number of debug log records that will be saved in this org. If the upper limit is hit, older debug logs will be deleted to make room for new ones.</p> -->
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/debugLogStep/debugLogStep.html:19]: 🪓

File: force-app/main/default/lwc/editorConfigStep/editorConfigStep.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import { LightningElement } from 'lwc'
+
+export default class OutboundConnectionStep extends LightningElement {
+    loading = false
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/editorConfigStep/editorConfigStep.js:4]: ✅

File: force-app/main/default/lwc/debugLogStep/debugLogStep.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import { LightningElement, api, track } from 'lwc'
+
+export default class DebugLogStep extends LightningElement {
+
+    @track logLevel = 'DEBUG'
+    @track logCount = 0
+    @track logLimit = 10000
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/debugLogStep/debugLogStep.js:7]: ✅

File: force-app/main/default/lwc/debugLogStep/debugLogStep.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+    get options() {
+        return [
+            { label: 'All Activity', value: 'DEBUG' },
+            { label: 'Errors Only', value: 'ERROR' }
+        ]
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/debugLogStep/debugLogStep.js:13]: ✅

File: force-app/main/default/lwc/customPropertyEditor/customPropertyEditor.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+    @api objectAPI;
+    @api recordID;
+    @api operation;
+
+    picklistOptions = [
+        {
+            label: '1',
+            value: '1'
+        }
+    ]
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/customPropertyEditor/customPropertyEditor.js:14]: ✅

File: force-app/main/default/lwc/editorConfigStep/editorConfigStep.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/editorConfigStep/editorConfigStep.js:4]: 🪓

File: force-app/main/default/lwc/errorRetrySchedulerStep/errorRetrySchedulerStep.css [hunk 12]
```diff
@@ -0,0 +1,4 @@
+.bss-tooltip {
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/errorRetrySchedulerStep/errorRetrySchedulerStep.css:1]: See above 🪓

File: force-app/main/default/lwc/errorRetrySchedulerStep/errorRetrySchedulerStep.js [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+    @track modalButtonVariant;
+    @track modalButtonLabel;
+    @track isActive = false;
+    @track value = '';
+
+    openBatchModal() {
+        if(this.isActive) {
+            this.modalTitle = 'Deactivate Job';
+            this.modalButtonVariant = 'destructive'
+            this.modalButtonLabel = 'Deactivate'
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/errorRetrySchedulerStep/errorRetrySchedulerStep.js:16]: ✅ ✅ ✅

File: force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.css [hunk 14]
```diff
@@ -0,0 +1,17 @@
+.rl-vertical_nav {
+    max-width: 180px;
+    min-width: 180px;
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.css:3]: This was behaving peculiarly with both rem units and setting a single width, so I just put a slds-size class on the vertical nav instead – worked out better for my purposes anyway

File: force-app/main/default/lwc/syncSchedulerStep/syncSchedulerStep.css [hunk 15]
```diff
@@ -0,0 +1,4 @@
+.bss-tooltip {
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/syncSchedulerStep/syncSchedulerStep.css:1]: See above

File: force-app/main/default/lwc/syncSchedulerStep/syncSchedulerStep.js [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+        }
+    ]
+
+    updateValue(event){
+        let value = event.detail.value;
+        let label = event.detail.label;
+
+        this.value = value;
+        this.label = label;
+    }
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/syncSchedulerStep/syncSchedulerStep.js:78]: ✅

File: force-app/main/default/lwc/packageLogConfigStep/packageLogConfigStep.js [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+    @track capturedActivityType = [
+        {
+            label:'Errors Only',
+            value:'errorsOnly'
+        },
+        {
+            label:'All Activity',
+            value:'allActivity'
+        }
+    ]
```
[the remark this answers is not in the document]
Comment (kathleenconway) [reply force-app/main/default/lwc/packageLogConfigStep/packageLogConfigStep.js:18]: ✅

File: force-app/main/default/lwc/outboundConnectionStep/outboundConnectionStep.js (hunk 1 above)
Comment (kathleenconway) [reply force-app/main/default/lwc/outboundConnectionStep/outboundConnectionStep.js:4]: ✅

File: force-app/main/default/lwc/batchSchedulerStep/batchSchedulerStep.html (hunk 2 above)
Comment (kathleenconway) [reply force-app/main/default/lwc/batchSchedulerStep/batchSchedulerStep.html:44]: 🪓

File: force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.js (hunk 3 above)
Comment (kathleenconway) [reply force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.js:49]: 🤦‍♀️

File: force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.js (hunk 4 above)
Comment (kathleenconway) [reply force-app/main/default/lwc/fieldMappingStep/fieldMappingStep.js:6]: I think this should be readable by humans other than myself now – let me know if anything needs to move around

**Comment by kathleenconway** [comment]
- Cleaned out commented-out code & superfluous css classes
- Added semicolons to ends of statements
- Got rid of static sizing on vertical nav of mapping step
- Major rehab of fieldMappingStep.js

**Review by sean-appiphony — APPROVED** [review]

**Pull request merged**